File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] ans = new int[2];
        HashMap<Integer, Integer> mp = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int a = nums[i];
            int b = target - a;
            if (mp.containsKey(b)) {
                ans[0] = i;
                ans[1] = mp.get(b);
                return ans;
            }
            mp.put(a, i);
        }
        return new int[] {};
    }
}

File: Leetcode/1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int, int> mp;
        for (int i = 0; i < n; i++) {
            int val = nums[i];
            int req = target - val;
            if (mp.find(req) != mp.end()) {
                int idx = mp[req];
                return {idx, i};
            }
            mp[val] = i;
        }
        return {-1, -1};
    }
};